test/sequential/test-repl-persistent-history.js

```diff
@@ -0,0 +1,193 @@
+'use strict';
+
+// Flags: --expose-internals
+
+const common = require('../common');
+const stream = require('stream');
+const REPL = require('internal/repl');
+const assert = require('assert');
+const fs = require('fs');
+const util = require('util');
+const path = require('path');
+const os = require('os');
+
+common.refreshTmpDir();
+
+// Mock os.homedir()
+os.homedir = function() {
+  return common.tmpDir;
+};
+
+// Create an input stream specialized for testing an array of actions
+class ActionStream extends stream.Stream {
```

Comment:
For a test, this is fine, but generally subclassing a stream constructor makes me a liiiiittle nervous.